solarforecastarbiter/reports/figures/plotly_figures.py

```diff
@@ -0,0 +1,375 @@
+
+    x_offset = None
+
+    # Special handling for x-axis with dates
+    if category == 'weekday':
+        x_ticks = calendar.day_abbr[0:]
+
+    elif category == 'hour':
+        x_ticks = list(range(25))
+        # plotly's offset of 0, makes the bars left justified at the tick
+        x_offset = 0
+    else:
+        x_ticks = None
+
+    y_data = np.asarray(data['value'])
+    if len(y_data) == 0:
+        y_range = (None, None)
+    else:
+        y_min = np.nanmin(y_data)
+        y_max = np.nanmax(y_data)
+        y_range = calc_y_start_end(y_min, y_max)
+
+    unique_names = np.unique(np.asarray(data['name']))
+    palette = [next(palette) for _ in unique_names]
+    for i, name in enumerate(unique_names):
+        plot_data = data[data['name'] == name]
+        if len(plot_data['index']):
+            x_values = plot_data['index']
+        else:
+            x_values = []
+        if category == 'weekday':
+            # Fill with mon-fri values and pass to enforce displaying the full
+            # week of data.
+            y_values = [plot_data[plot_data['index'] == day]['value'].iloc[0]
+                        if not plot_data[plot_data['index'] == day].empty
+                        else np.nan for day in x_ticks]
+            x_values = x_ticks
+        else:
+            y_values = plot_data['value']
+        # Create figure
+        title = name + ' ' + metric_name
+        fig = go.Figure()
+        fig.add_trace(go.Bar(x=x_values, y=y_values, offset=x_offset,
+                             marker=go.bar.Marker(color=palette[i])))
+
+        fig.update_layout(
+            title=f'<b>{title}</b>',
+            xaxis_title=x_axis_label,
+            yaxis_title=y_axis_label,
+            **PLOT_LAYOUT_DEFAULTS)
+        configure_axes(fig, x_ticks, y_range)
+        figs[name] = fig
+    return figs
+
+
+def nested_bar():
+    raise NotImplementedError
+
+
+def joint_distribution():
+    raise NotImplementedError
+
+
+def marginal_distribution():
+    raise NotImplementedError
+
+
+def taylor_diagram():
+    raise NotImplementedError
+
+
+def probabilistic_timeseries():
+    raise NotImplementedError
+
+
+def reliability_diagram():
+    raise NotImplementedError
+
+
+def rank_histogram():
+    raise NotImplementedError
+
+
+def output_svg(fig):
+    """
+    Generates an SVG from the Plotly figure. Errors in the process are logged
+    and an SVG with error text is returned.
+
+    Parameters
+    ----------
+    fig : plotly.graph_objects.Figure
+
+    Returns
+    -------
+    svg : str
+
+    Notes
+    -----
+    Requires `Orca <https://plot.ly/python/orca-management/>`_ for generating
+    svgs. If orca is not installed, an svg with an error message will be
+    returned.
+    """
+    try:
+        svg = fig.to_image(format='svg').decode('utf-8')
+    except Exception:
+        logger.error('Could not generate SVG for figure %s',
+                     getattr(fig, 'name', 'unnamed'))
```

Comment:
I guess plotly figures don't have name? can you replace name with something that identifies the figure?